src/EditorFeatures/CSharpTest/ConvertAutoPropertyToFullProperty/ConvertAutoPropertyToFullPropertyTests.cs

```diff
@@ -938,6 +939,36 @@ public int Goo
         }
     }
 }
+";
+            await TestInRegularAndScriptAsync(text, expected, options: UseUnderscorePrefixedFieldName);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ConvertAutoPropertyToFullProperty)]
+        public async Task NonStaticPropertyWithCustomStaticFieldName()
+        {
+            var text = @"
+class goo
```

Comment:
💡 It seems unintentional for the class name to match the name of the generated field. Consider renaming this to `class TestClass` to ensure the test continues to work without alterations if in the future we choose a better name for the generated field if a collision like this occurs.

💡 Consider filing a new issue to handle this situation without introducing a naming collision (e.g. create the field `gooValue` or `goo1` or whatever we use for disambiguation during name suggestion).